pkg/kubelet/pleg/generic.go

```diff
@@ -218,36 +226,59 @@ func (g *GenericPLEG) relist() {
 		needsReinspection = make(map[types.UID]*kubecontainer.Pod)
 	}
 
+	// update PodCache and PodRecords
+	g.updatePodCacheByWorkers(eventsByPodID, needsReinspection)
```

Comment:
`updatePodCacheByWorkers` also updates podRecords, I will change the function name.